File: frontend/src/app/services/models/reservation-details-client-dto.ts
/* tslint:disable */
/* eslint-disable */
import { FlightDetailsClientDto } from '../models/flight-details-client-dto';
export interface ReservationDetailsClientDto {
  cabinClass?: 'FIRST' | 'BUSINESS' | 'ECONOMY';
  flightDto?: FlightDetailsClientDto;
  id?: string;
  price?: number;
  reservationDate?: string;
  seatNumber?: number;
  status?: 'PENDING' | 'ACTIVE' | 'CANCELLED' | 'EXPIRED' | 'COMPLETED' | 'FAILED';
}

File: src/main/java/FlyAway/reservation/dto/ReservationDetailsClientDto.java
package FlyAway.reservation.dto;

import FlyAway.flight.aircraft.CabinClass;
import FlyAway.flight.dto.FlightDetailsClientDto;
import FlyAway.reservation.ReservationStatus;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

public record ReservationDetailsClientDto(
        UUID id,
        ReservationStatus status,
        CabinClass cabinClass,
        Integer seatNumber,
        BigDecimal price,
        LocalDateTime reservationDate,
        FlightDetailsClientDto flightDto

) {
}
